addons/popochiu/editor/main_dock/popochiu_row/object_row/popochiu_object_row.gd

```diff
@@ -0,0 +1,363 @@
+		"" if audio_files.is_empty()
+		else " ([b]%d[/b] audio cues will be deleted)" % audio_files.size()
+	]
+	_delete_dialog.on_confirmed = _remove_from_core
+	
+	PopochiuEditorHelper.show_delete_confirmation(_delete_dialog)
+
+
+func _search_audio_files(dir: EditorFileSystemDirectory) -> Array:
+	var files := []
+	
+	for idx in dir.get_subdir_count():
+		files.append_array(_search_audio_files(dir.get_subdir(idx)))
+	
+	for idx in dir.get_file_count():
+		match dir.get_file_type(idx):
+			"AudioStreamOggVorbis", "AudioStreamMP3", "AudioStreamWAV":
+				files.append(dir.get_file_path(idx))
+	
+	return files
+
+
+func _remove_from_core() -> void:
+	# Check if the files should be deleted in the file system
+	if _delete_dialog.check_box.button_pressed:
+		_delete_from_file_system()
+	elif type in PopochiuResources.MAIN_TYPES:
+		show_add_to_core()
+	
+	var edited_scene: Node = EditorInterface.get_edited_scene_root()
+	if edited_scene and edited_scene.get("script_name") and edited_scene.script_name == name:
+		# If the open scene matches the object being deleted, skip saving the scene
+		queue_free()
+		return
+	
+	EditorInterface.save_scene()
+	queue_free()
+
+
+## Remove this object's directory (subfolders included) from the file system.
+func _delete_from_file_system() -> void:
+	var object_dir: EditorFileSystemDirectory = \
+		EditorInterface.get_resource_filesystem().get_filesystem_path(path.get_base_dir())
+	
+	# Remove files, sub folders and its files.
+	_recursive_delete(object_dir)
+
+
+## Remove the `dir` directory from the system. For Godot to be able to delete a directory, it has to
+## be empty, so this method first deletes the files from from the directory and each of its
+## subdirectories.
+func _recursive_delete(dir: EditorFileSystemDirectory) -> void:
+	if dir.get_file_count() > 0:
+		assert(
+			_delete_files(dir) == OK,
+			"[Popochiu] Error removing files in recursive elimination of %s" % dir.get_path()
+		)
+	
+	if dir.get_subdir_count() > 0:
+		for folder_idx in dir.get_subdir_count():
+			# Check if there are more folders inside the folder or delete the files inside it before
+			# deleting the folder itself
+			_recursive_delete(dir.get_subdir(folder_idx))
+	
+	assert(
+		 DirAccess.remove_absolute(dir.get_path()) == OK,
+		"[Popochiu] Error removing folder in recursive elimination of %s" % dir.get_path()
+	)
+	EditorInterface.get_resource_filesystem().scan()
+
+
+## Delete files within [param dir] directory. First, get the paths to each file, then delete them
+## one by one calling [method EditorFileSystem.update_file], so that in case it's an imported file,
+## its [b].import[/b] is also deleted.
+func _delete_files(dir: EditorFileSystemDirectory) -> int:
+	# This array will store the paths of the files to be deleted.
+	var files_paths := []
+	var deleted_audios := []
+	
+	for file_idx: int in dir.get_file_count():
+		match dir.get_file_type(file_idx):
+			"AudioStreamOggVorbis", "AudioStreamMP3", "AudioStreamWAV":
```

Comment:
It's the second time you do this test, I would bring this in a constant or a function, so that adding formats would not require going around the code in search for this list.

Reply:
Sure. I'll do.

Reply:
Done.